tests/stackage_zlib_runpath/BUILD.bazel

```diff
@@ -0,0 +1,122 @@
+load(
+    "//tests:inline_tests.bzl",
+    "py_inline_test",
+)
+load("dynamic_libraries.bzl", "dynamic_libraries")
+
+dynamic_libraries(
+    name = "libz",
+    srcs = ["@zlib.dev//:zlib"],
+    filter = "libz",
+    solib_names = "libz_soname",
+    tags = ["requires_zlib"],
+)
+
+dynamic_libraries(
+    name = "libHSzlib",
+    srcs = ["@stackage-zlib//:zlib"],
+    filter = "libHSz",
+    tags = ["requires_zlib"],
+)
+
+# This test case tests that haskell_cabal_library will generate a relative
+# RUNPATH entry for the dependency on the nixpkgs provided libz. Relative
+# meaning an entry that starts with $ORIGIN (Linux) or @loader_path (MacOS).
+# The alternative is an absolute path, which would be wrong for the nixpkgs
+# provided libz, as we want the RUNPATH entry to point to Bazel's _solib_<cpu>
+# directory and its absolute path depends on the output root or execroot.
+#
+# It uses :libz_soname generated above to determine the expected RUNPATH entry
+# for the libz dependency. The :libz_soname file will contain the file names of
+# the libz library files underneath the `_solib_<cpu>` directory.
+#
+# It uses :libHSzlib to access the dynamic library output of
+# haskell_cabal_library and read the RUNPATH entries.
+#
+# Note, ideally we would test that haskell_cabal_library _only_ generates a
+# relative RUNPATH entry and no absolute entries that leak the execroot into
+# the cache. Unfortunately, haskell_cabal_library generates such an entry at
+# the moment. See https://github.com/tweag/rules_haskell/issues/1130.
+py_inline_test(
+    name = "stackage_zlib_runpath",
+    args = [
+        "$(rootpath :libz_soname)",
```

Comment:
are `solib_names` exported as targets?

Reply:
It's an [output attribute](https://docs.bazel.build/versions/master/skylark/lib/attr.html#output) that means that it becomes accessible as a label.